org.adempiere.base/src/org/compiere/model/MTable.java

```diff
@@ -1103,4 +1104,29 @@ public boolean hasCustomTree() {
 		return hasCustomTree.booleanValue();
 	}
 
+	/**
+	 * Get Partition Name of the table of the given level
+	 * @param tableName
+	 * @param primaryLevelOnly - if true, ignore the sub-partition, if exists
+	 * @return table partition name, or empty
+	 */
+	public static String getPartitionName(Properties ctx, String tableName, boolean primaryLevelOnly, String trxName) {
+		// collect partition key columns
+		String[] partitionColsAll = MTablePartition.getPartitionKeyColumns(ctx, tableName, trxName);
+		int level = primaryLevelOnly ? 1 : partitionColsAll.length;			
+		ArrayList<Object> partitionCols = new ArrayList<>();
+		for(int i = 0; i < level; i++) {
+			partitionCols.add(partitionColsAll[i]);
+		}

```

Comment:
this is totally unnecessary, you can directly do a for loop of partitionColsAll. i.e

`for(int i = 0; i < level; i++) partitionName.append("_").append(partitionColsAll[i])`